Created the basic code for the driver controlled swerve drive

ethanharjabrata opened the pull request:
Some values need updating:
Motor channels
Encoder channels
DistancePerPulse

ethanharjabrata committed:
Created the basic code for the driver controlled swerve drive

Some values need updating:
Motor channels
Encoder channels
DistancePerPulse

ethanharjabrata committed:
Test code

Code used to find out the distance per pulse of the robot

WebCoder49 committed:
Make swerve drive code slightly more readable

WebCoder49 reviewed (approved):
LGTM

WebCoder49 committed:
Merge pull request #12 from brightonfrc/main

Make build.gradle errorless

WebCoder49 committed:
Merge pull request #15 from brightonfrc/7-driver-controlled-swerve-drive

Add swerve drive subsystems

ethanharjabrata committed:
Draft

Created the code for the motors, but I haven't bound the snap function to a button yet.

ethanharjabrata committed:
Code for the snap Command

It is bound to button 1 on the Joystick

ethanharjabrata committed:
New commit

Includes code so that the robot can update its PID.
Also allows the robot to offset the encoders such that they will be field oriented (assuming that the robot starts facing the field)

ethanharjabrata committed:
This should work

ethanharjabrata committed:
Sorry

Forgot to update the robotContainer

WebCoder49 reviewed (changes requested):
Thank you for this! However, please see my PID tuning comment.

ethanharjabrata committed:
new update to use the proper motors

ethanharjabrata committed:
change to the motor subsystem

WebCoder49 committed:
Get PID from constants for now

ethanharjabrata committed:
Merge branch '14-snap-to-90degree-increment-angles' into 7-driver-controlled-swerve-drive

ethanharjabrata committed:
Merge pull request #19 from brightonfrc/7-driver-controlled-swerve-drive

7 driver controlled swerve drive

ethanharjabrata committed:
added the port numbers to the robot container

personofalltime committed:
CAN IDs Assigned

WebCoder49 committed:
Get encoders from SparkMAX; fix some naming conventions

ethanharjabrata committed:
A quick update

Changed it so that the new bearing will only update if the joystick points in a direction that would probably result in a new snap heading.

ethanharjabrata committed:
Ignore this

ethanharjabrata committed:
also ignore this

ethanharjabrata committed:
Fixing a logic error

I changed it such that the encoder will actually tell you the distance moved since the last time it was called, not the total distance moved as this would be problematic if you were using two different commands during that time.
I have also used two different constants,
distancePerRotation meaning the circumference of the motor.
movementPerRotation meaning the actual distance travelled by the robot when the move motors rotate once.

WebCoder49 committed:
Merge pull request #24 from brightonfrc/7-driver-controlled-swerve-drive

Update snap-to-90deg code with swerve drive

ethanharjabrata committed:
new testing version

ethanharjabrata committed:
test config

ethanharjabrata committed:
Working version

ethanharjabrata committed:
field oriented version

ethanharjabrata committed:
added some extra stuff to SmartDashBoard

ethanharjabrata committed:
Snap Update

ethanharjabrata committed:
updated the constants

ethanharjabrata committed:
update to encoder code

ethanharjabrata committed:
added more smartdashboard stuff

ethanharjabrata committed:
Testing update

ethanharjabrata committed:
fixing bugs I noticed while testing snap

ethanharjabrata committed:
old version

reload this to use the robotBearingController that has a continuous input

ethanharjabrata committed:
Working version

ethanharjabrata committed:
Another quick update to errors I noticed while testing snap

ethanharjabrata committed:
Updating drive.java to the working version

ethanharjabrata committed:
added javadocs for readability

WebCoder49 committed:
Merge branch '14-snap-to-90degree-increment-angles' into 7-driver-controlled-swerve-drive
